File: Python/treehopper/libraries/sensors/optical/AmbientLight.py
from abc import abstractmethod
from treehopper.libraries.sensors import Pollable


class AmbientLight(Pollable):
    def __init__(self):
        super().__init__()
        self._lux = 0

    @property
    def lux(self):
        if self.auto_update_when_property_read:
            self.update()

        return self._lux

    @abstractmethod
    def update(self):
        pass

File: Java/api/treehopper.libraries/src/main/java/io/treehopper/libraries/sensors/optical/AmbientLight.java
package io.treehopper.libraries.sensors.optical;

public abstract class AmbientLight implements IAmbientLight {
    protected double lux;
    private boolean autoUpdateWhenPropertyRead = true;

    @Override
    public double getLux() {
        if (isAutoUpdateWhenPropertyRead())
            update();

        return lux;
    }

    @Override
    public boolean isAutoUpdateWhenPropertyRead() {
        return autoUpdateWhenPropertyRead;
    }

    @Override
    public void setAutoUpdateWhenPropertyRead(boolean value) {
        autoUpdateWhenPropertyRead = value;
    }
}
